### PR TITLE
User info

### PR DESCRIPTION
**Pull request by cavis**
Fetch user info from the server side, using the ID JWT.  Cache into the session.

In support of PRX/augury.prx.org#472.  Copying all the style/javascript-y parts of the app picker move over into the parent app with this.  This gem just gets you the userinfo from ID, and you display how you want.  (Which will also help with building out the new sidenav designs).

### CODE REVIEW

File: lib/prx_auth/rails/ext/controller.rb [hunk 1]
```diff
@@ -34,10 +35,36 @@ def authenticate!
         redirect_to PrxAuth::Rails::Engine.routes.url_helpers.new_sessions_path
       end
 
+      def prx_auth_needs_refresh?(jwt_ttl)
+        request.get? && jwt_ttl < PRX_JWT_REFRESH_TTL
```
Comment (cavis) [inline lib/prx_auth/rails/ext/controller.rb:41]: This is the check for "should we interrupt the request cycle to refresh our JWT?"

Moved it into its own method so Augury can override.  And ensure we're not in the middle of a Turbo `fetch`.  As currently, a non-full-page Turbo request can't successfully follow the redirect (CORS issues).  Eventually, an upstream Turbo bug/feature may allow us to just break out of Turbo entirely and force the refresh.

File: lib/prx_auth/rails/ext/controller.rb [hunk 2]
```diff
@@ -34,10 +35,36 @@ def authenticate!
...
+      end
+
       def current_user
         prx_auth_token
       end
 
+      def current_user_info
+        session[PRX_USER_INFO_SESSION_KEY] ||= fetch_userinfo
```
Comment (cavis) [inline lib/prx_auth/rails/ext/controller.rb:49]: As I was writing this, I started thinking ... we are really depending on non-cookie session storage now.  The JWT alone is going to overfill it.  I wonder if that should/could be built into the gemspec dependencies?

File: lib/prx_auth/rails/ext/controller.rb [hunk 3]
```diff
@@ -34,10 +35,36 @@ def authenticate!
...
+          label = name.sub(/^https?:\/\//, '').sub(/\..+/, '').capitalize
+          ["PRX #{label}", url]
+        end
+
+        # only return entire list in development
+        if ::Rails.env.production? || ::Rails.env.staging?
+          apps.to_h.select { |k, v| v.match?(/\.(org|tech)/) }
+        else
+          apps.to_h
+        end
```
Comment (cavis) [inline lib/prx_auth/rails/ext/controller.rb:67]: All this logic stolen from the ID javascript.  Maybe we'll want to change it someday ... but there are tests now!

**Review by svevang — APPROVED** [review]
👍

**Pull request merged**